node/src/storage_mngr.rs

```diff
@@ -70,23 +94,24 @@ where
         .map(|(k, v)| Ok((serialize(k)?, serialize(v)?)))
         .collect();
 
-    match kv_bytes {
-        Ok(kv_bytes) if kv_bytes.is_empty() => Either::B(futures::future::finished(())),
-        Ok(kv_bytes) => Either::A(addr.send(PutBatch(kv_bytes)).flatten()),
-        Err(e) => Either::B(futures::future::failed(e)),
+    async move {
+        match kv_bytes {
+            Ok(kv_bytes) if kv_bytes.is_empty() => Ok(()),
+            Ok(kv_bytes) => addr.send(PutBatch(kv_bytes)).await?,
+            Err(e) => Err(e),
+        }
     }
 }
 
 /// Delete value associated to key
-pub fn delete<K>(key: &K) -> impl Future<Item = (), Error = failure::Error>
+pub async fn delete<K>(key: &K) -> Result<(), failure::Error>
```

Comment:
This function is not used, but it needs to be rewritten to be similar to `get`. Because using `async fn` will result in lifetime errors if the output future lives longer than the input key.